olp-cpp-sdk-core/include/olp/core/client/HttpResponse.h

```diff
@@ -33,7 +34,7 @@ namespace client {
 class CORE_API HttpResponse {
  public:
   HttpResponse() = default;
-  HttpResponse(int status, std::string response = std::string())
+  HttpResponse(int status, std::shared_ptr<std::stringstream> response = nullptr)
```

Comment:
can we remove shared_ptr and pass std::stringstream directly with std::move?
alternatively, we can also have a constructor with a std::string

Reply:
Currently, user creates the stream and passes it to us. moving the stream would discard user's instance of it. I doubt we can afford that right now.
See usage example in NetworkMock

Reply:
We could use moves if the stream was created by us and passed as an outcome, but since that's a user-initialized value, it's too risky to move it.